packages/@expo/cli/src/start/server/middleware/CorsMiddleware.ts

```diff
@@ -31,9 +53,13 @@ export function createCorsMiddleware(exp: ExpoConfig) {
           )
         );
         return;
+      } else if (!isLocalhost && isAllowedHost) {
```

Comment:
This changes the original behaviour. We're not adding the header for localhost hostnames. However, this also changes the behaviour to not add this header when we're getting arbitrary requests.

Previously, due to the above `!isSameOrigin && !isAllowedHost` logical-and, we were always mirroring the `Origin` header otherwise. However, it seems like the intention is not to add this header if we're not dealing with an allowed host input

cc @Kudo: Does this change break any of our prior assumptions? It's not clear to me if always adding this was intentional